### PR TITLE
[New Rule] GCP Firewall Rule Created

### LINKED ISSUE #282: [New Rule] GCP Firewall Rule Created
## Description

Identifies when a firewall rule is created in Google Cloud Platform (GCP). Virtual Private Cloud (VPC) firewall rules can be configured to allow or deny connections to or from virtual machine (VM) instances. An adversary may create a new firewall rule in order to weaken their target's security controls and allow more permissive ingress or egress traffic flows for their benefit.

```
event.module:googlecloud and event.dataset:googlecloud.audit and event.action:v*.compute.firewalls.insert
```

## Required Info

- **Eventing Sources:**

`filebeat-*`

- **Platforms**

GCP

- **Target ECS Version:** `1.6.0`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
  - https://cloud.google.com/vpc/docs/firewalls

## Example Data

Note, the `event.outcome` was `unknown` when I created, modified, and deleted GCP firewall rules.

image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves #282 

## Summary
Identifies when a firewall rule is created in Google Cloud Platform (GCP). Virtual Private Cloud (VPC) firewall rules can be configured to allow or deny connections to or from virtual machine (VM) instances. An adversary may create a new firewall rule in order to weaken their target's security controls and allow more permissive ingress or egress traffic flows for their benefit.

I'd like to include the [Impair Defenses](https://attack.mitre.org/techniques/T1562/) ATT&CK technique in this rule, but I think our schema might need to be updated before I can do this.

I found it interesting that the `event.outcome` is `unknown` when creating, modifying, or deleting firewall rules. I did not see a `success` value like many other GCP events. I decided to leave `event.outcome` out of this rule.

image

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by threat-punter** [comment]
Blocked until I can add ATT&CK technique T1562 to the rule. cc @brokensound77

**Review by brokensound77 — APPROVED** [review]
LGTM 👍 

Once #330 merges (pending @rw-access review 🙂 ), this should accept the technique and pass validation

File: rules/gcp/defense_evasion_gcp_firewall_rule_created.toml
Comment (bm11100) [inline rules/gcp/defense_evasion_gcp_firewall_rule_created.toml:24]:
```suggestion
name = "GCP Firewall Rule Creation"
```

**Review by bm11100 — COMMENTED** [review]
🤖

**Review by bm11100 — APPROVED** [review]

**Pull request merged**
